IJoypadEventDriven deprecated

**Describe the bug**
When clean building yarp, and invoking `IJoypadController` from a walking module the following error apears:
```
In file included from /usr/local/src/robot/yarp/build/src/libYARP_conf/include/yarp/conf/api.h:13:0,
                 from /usr/local/src/robot/yarp/src/libYARP_OS/include/yarp/os/api.h:13,
                 from /usr/local/src/robot/yarp/src/libYARP_OS/include/yarp/os/LogStream.h:16,
                 from /usr/local/src/robot/element_capture-point-walking/code/cpp/Joypad_module/src/main.cpp:10:
/usr/local/src/robot/yarp/src/libYARP_dev/include/yarp/dev/IJoypadController.h:257:33: error: expected unqualified-id before ‘[’ token
     explicit IJoypadEventDriven(YARP_DEPRECATED_MSG("Use IJoypadEventDriven(double)") int rate);
                                 ^
/usr/local/src/robot/yarp/src/libYARP_dev/include/yarp/dev/IJoypadController.h:257:33: error: expected ‘)’ before ‘[’ token
Joypad_module/CMakeFiles/WalkingJoypadModule.dir/build.make:62: recipe for target 'Joypad_module/CMakeFiles/WalkingJoypadModule.dir/src/main.cpp.o' failed
make[2]: *** [Joypad_module/CMakeFiles/WalkingJoypadModule.dir/src/main.cpp.o] Error 1
```
However if i remove the `YARP_DEPRECATED_MSG` from the corresponding file and build yarp again, and then re-build my walking module, the error doesn't appear.

**To Reproduce**
* You have to have the `element_capture-point-walking` module, precisely the `virtualizer/CVirt` Fork/branch. or any code that uses that class.
* Build the source code and you will get the error.

**Configuration (please complete the following information):**
 - OS: ubuntu
 - yarp version: latest devel 3.0.0
-  compiler: gcc 5.4.0

@aerydna

--- a/doc/release/v3_0_1.md
+++ b/doc/release/v3_0_1.md
@@ -61,7 +61,10 @@ Bug Fixes
 
 #### YARP_dev
 
-* Fixed getCurrent/getCurrents. Now the `stateExt` port is used, and the methods are called through the `iCurrentControl` interface.
+* Fixed getCurrent/getCurrents. Now the `stateExt` port is used, and the methods
+  are called through the `iCurrentControl` interface.
+* Fixed deprecation message placement for the constructor of
+  `IJoypadEventDriven`. It was failing enabling c++14 compiling with gcc(#1747).
 
 #### YARP_math
 

--- a/src/libYARP_dev/include/yarp/dev/IJoypadController.h
+++ b/src/libYARP_dev/include/yarp/dev/IJoypadController.h
@@ -254,7 +254,12 @@ public:
 
     IJoypadEventDriven();
 #ifndef YARP_NO_DEPRECATED // Since YARP 3.0.0
+#if defined(_MSC_VER) && _MSC_VER <= 1900 // VS 2015
     explicit IJoypadEventDriven(YARP_DEPRECATED_MSG("Use IJoypadEventDriven(double)") int rate);
+#else
+    YARP_DEPRECATED_MSG("Use IJoypadEventDriven(double)")
+    explicit IJoypadEventDriven(int rate);
+#endif
 #endif
     explicit IJoypadEventDriven(double period);
 

